facebook/delphiFacebook/R/contingency_variables.R

```diff
@@ -1452,3 +1561,237 @@ code_addl_activities <- function(input_data, wave) {
+    input_data$race_asian <- input_data$D7 == 2
+    input_data$race_black_african_american <- input_data$D7 == 3
+    input_data$race_native_hawaiian_pacific_islander <- input_data$D7 == 4
+    input_data$race_white <- input_data$D7 == 5
+    input_data$race_multiple_other <- (input_data$D7 == 6 | grepl(",", input_data$D7))
+  } else {
+    input_data$race_american_indian_alaska_native <- NA
+    input_data$race_asian <- NA
+    input_data$race_black_african_american <- NA
+    input_data$race_native_hawaiian_pacific_islander <- NA
+    input_data$race_white <- NA
+    input_data$race_multiple_other <- NA
+  }
+
+  if ("D8" %in% names(input_data)) {
+    input_data$education_less_than_highschool <- input_data$D8 == 1
+    input_data$education_highschool_or_equivalent <- input_data$D8 == 2
+    input_data$education_some_college <- input_data$D8 == 3
+    input_data$education_2yr_degree <- input_data$D8 == 4
+    input_data$education_4yr_degree <- input_data$D8 == 5
+    input_data$education_masters <- input_data$D8 == 8
+    input_data$education_professional_degree <- input_data$D8 == 6
+    input_data$education_doctorate <- input_data$D8 == 7
+  } else {
+    input_data$education_less_than_highschool <- NA
+    input_data$education_highschool_or_equivalent <- NA
+    input_data$education_some_college <- NA
+    input_data$education_2yr_degree <- NA
+    input_data$education_4yr_degree <- NA
+    input_data$education_masters <- NA
+    input_data$education_professional_degree <- NA
+    input_data$education_doctorate <- NA
+  }
+
+  if ("D12" %in% names(input_data)) {
+    input_data$language_home_english <- input_data$D12 == 1
+    input_data$language_home_spanish <- input_data$D12 == 2
+    input_data$language_home_chinese <- input_data$D12 == 3
+    input_data$language_home_vietnamese <- input_data$D12 == 4
+    input_data$language_home_french <- input_data$D12 == 5
+    input_data$language_home_portugese <- input_data$D12 == 6
+    input_data$language_home_other <- input_data$D12 == 7
+  } else {
+    input_data$language_home_english <- NA
+    input_data$language_home_spanish <- NA
+    input_data$language_home_chinese <- NA
+    input_data$language_home_vietnamese <- NA
+    input_data$language_home_french <- NA
+    input_data$language_home_portugese <- NA
+    input_data$language_home_other <- NA
+  }
+
+  # Children by age
+  if (all(c("E1_1", "E1_2", "E1_3", "E1_4") %in% names(input_data))) {
+    # All subquestions coded as 1 = Yes, 2 = No, 5 = don't know
+    input_data$children_prek <- input_data$E1_1 == 1
+    input_data$children_gr1_5 <- input_data$E1_2 == 1
+    input_data$children_gr6_8 <- input_data$E1_3 == 1
+    input_data$children_gr9_12 <- input_data$E1_4 == 1
+  } else {
+    input_data$children_prek <- NA
+    input_data$children_gr1_5 <- NA
+    input_data$children_gr6_8 <- NA
+    input_data$children_gr9_12 <- NA
+  }
+
+  if ("E3" %in% names(input_data)) {
```

Comment:
For E3, do we want to treat "I don't know" as NA, so it's among people who know?